### DISCUSSION TITLE
How do you perform an INTERSECT query using Supabase's JavaScript client?

### DISCUSSION THREAD

**Discussion by vossavant** [supabase/supabase#22140] [Questions] [answered]
I am using Supabase in a Nuxt 3 app and cannot find documentation on how to perform an `INTERSECT` query. Normal SQL is not a problem for me and returns the expected rows in Supabase's SQL Editor:

```
SELECT part_number, description
FROM product
INTERSECT
SELECT part_number, description
FROM product_temp
```

I have regularly been at a loss for how to "convert" a non-standard query like this to Supabase's JavaScript syntax. For instance, a regular SELECT query looks like:
```
const { data, error } = await supabase
  .from('countries')
  .select()
```

How would I perform the INTERSECT query at the beginning of my question using the JavaScript syntax directly above?

**Comment by GaryAustin1** [comment, marked as answer]
For more advanced queries you need to use a view or an rpc call to a Postgres function.  https://postgrest.org/en/v12/references/api/tables_views.html Has details of what can be done with the REST API.

**Reply by vossavant** [reply]
Thanks for pointing me to Postgres functions. I ended up writing this in the Supabase SQL editor:
```
create or replace function get_duplicates() 
returns setof record as $$
SELECT part_number, vendor_id, pricelist_id
FROM product
INTERSECT
SELECT part_number, vendor_id, pricelist_id
FROM product_temp
$$ language sql
```

Then, in my JavaScript:
```
const { data, error } = await supabase.rpc('get_duplicates')
```
